go/libkb/kbfs_folder.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2015 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package libkb
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+)
+
+var caseSensitiveServices = map[string]bool{"hackernews": true}
+
+// CanonicalizeName takes a folder name as input and returns a
+// canonical version of it. It also checks for basic errors.
+func CanonicalizeName(name string) (string, error) {
+	// check for invalid characters
+	if strings.ContainsAny(name, "/+") {
+		return "", errors.New("invalid character in name")
+	}
+
+	// split into individual ids
+	ids := strings.Split(name, ",")
+	for i, id := range ids {
```

Comment:
probably want to complain about empty names.

Also what about `#`-style readers?